### ISSUE TITLE
Wrong API version for automationAccounts/softwareUpdateConfigurations when deleting nested resource

### ISSUE THREAD

**Issue by bartwitkowski** [hashicorp/terraform-provider-azurerm#14810]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

- Terraform 1.0.10
- azurerm v2.85.0

### Affected Resource(s)

* `azurerm_resource_group_template_deployment`

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group_template_deployment" "windows-suc" {
  depends_on = [
    azurerm_virtual_machine.vms,
    azurerm_virtual_machine_extension.log-analytics-agent
  ]
  for_each = { 
      for key, value in var.servers: key => value if value.enable_schedule_update == true  
  }
  name                = "${var.project_shortname}-${each.key}-${var.env_shortname}-VM-suc"
  resource_group_name = var.resource_group_name_law
  deployment_mode     = "Incremental"
  parameters_content = jsonencode({
    "location" = {
      value = var.region
    }
    "aa_name" = {
      value = "${var.project_shortname}-AutomationAccount-${title(var.env_shortname)}"
    }
    "vm_name" = {
      value = "${var.project_shortname}-${each.key}-${var.env_shortname}-VM"
    }
    "vm_rg_name" = {
      value = var.servers_resource_group_name
    }
    "subscription_id" = {
      value = var.project_subscription_id
    }
    "week_days" = {
      value = each.value.updates_week_days
    }
    "start_date" = {
      value = substr(time_offset.next_day[each.key].rfc3339, 0, 10)
    }
    "start_time" = {
      value = each.value.updates_start_time
    }
    "reboot_option" = {
      value = each.value.reboot_option
    }
  })
  template_content = file("${path.module}/scripts/update-management.json")
}
```
The ARM deployment itself:

```json
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "location": {
            "type": "String",
            "metadata": {
                "description": "Location of the resource"
            }
        },
        "aa_name": {
            "type": "String",
            "metadata": {
                "description": "Name of the Automation Account"
            }
        },
        "vm_name": {
            "type": "String",
            "metadata": {
                "description": "Name of the VM"
            }
        },
        "vm_rg_name": {
            "type": "String",
            "metadata": {
                "description": "Name of the VM Resource Group"
            }
        },
        "subscription_id": {
            "type": "String",
            "metadata": {
                "description": "Project subscription ID"
            }
        },
        "week_days": {
            "type": "String",
            "metadata": {
                "description": "Week days on which the vm software update has to run"
            }
        },
        "start_date": {
            "type": "String",
            "metadata": {
                "description": "Set next day to apply schedule plan"
            }
        },
        "start_time": {
            "type": "String",
            "metadata": {
                "description": "Start time on which the vm software update has to start"
            }
        },
        "reboot_option": {
            "type": "String",
            "metadata": {
                "description": "Choose reboot option: IfRequired, Never, Always"
            }
        }
    },
    "resources": [
        {
            "type": "Microsoft.Automation/automationAccounts/softwareUpdateConfigurations",
            "apiVersion": "2017-05-15-preview",
            "location": "[parameters('location')]",
            "name": "[concat(parameters('aa_name'),'/',parameters('vm_name'))]",
            "properties": {
                "updateConfiguration": {
                    "operatingSystem": "Windows",
                    "duration": "PT2H0M",
                    "windows": {
                        "includedUpdateClassifications": "Critical,Security,UpdateRollup,Definition",
                        "rebootSetting": "[parameters('reboot_option')]"
                    },
                    "azureVirtualMachines": [
                        "[concat('/subscriptions/',parameters('subscription_id'),'/resourceGroups/',parameters('vm_rg_name'),'/providers/Microsoft.Compute/virtualMachines/',parameters('vm_name'))]"
                    ]
                },
                "scheduleInfo": {
                    "frequency": "Week",
                    "startTime": "[concat(parameters('start_date'),'T',parameters('start_time'))]",
                    "timeZone": "Europe/Copenhagen",
                    "interval": 1,
                    "advancedSchedule": {
                        "weekDays": [
                          "[parameters('week_days')]"
                        ]
                    }
                }
            }
        }
    ]
  }
```
### Expected Behaviour

When deleting VM with Update Management configured Terraform should delete softwareUpdateConfigurations with appropriate API version.

### Actual Behaviour

Deletion fails because of wrong API version **2021-06-22** chosen by Terraform. That API version seems to be automationAccounts and not the softwareUpdateConfigurations.
Looking at the ARM template exported from Azure the correct API is **2019-06-01**.

Short error output:
`Error: removing items provisioned by this Template Deployment: deleting Nested Resource "/subscriptions/xxx/resourceGroups/SOB-LogAnalytics-Dev-RG/providers/Microsoft.Automation/automationAccounts/SOB-AutomationAccount-Dev/softwareUpdateConfigurations/SOB-ETL1-dev-VM": resources.Client#DeleteByID: Failure sending request: StatusCode=400 -- Original Error: Code="NoRegisteredProviderFound" Message="No registered resource provider found for location 'westeurope' and API version '2021-06-22' for type 'automationAccounts/softwareUpdateConfigurations'. The supported api-versions are '2017-05-15-preview, 2018-01-15, 2018-06-30, 2019-06-01, 2020-01-13-preview'. The supported locations are 'japaneast, eastus2, westeurope, southafricanorth, ukwest, switzerlandnorth, brazilsoutheast, norwayeast, germanywestcentral, uaenorth, switzerlandwest, japanwest, uaecentral, australiacentral2, southindia, francesouth, norwaywest, westus3, swedencentral, southeastasia, southcentralus, northcentralus, eastasia, centralus, westus, australiacentral, australiaeast,
koreacentral, eastus, westus2, brazilsouth, uksouth, westcentralus, northeurope, canadacentral, australiasoutheast, centralindia, francecentral, jioindiawest, jioindiacentral, canadaeast, koreasouth'."`

### Steps to Reproduce

- Create VM with above `azurerm_resource_group_template_deployment` resource.
- Try to delete the VM
- The error will occur.

### References

There was identical problem previously: #10654
Solved in AzureRM provider version **2.71.0**
But, as you can see, still exists in **2.85.0**

**Comment by bartwitkowski** [comment]
Anyone?
I still have this problem.
Tried with AzureRM version 2.96.0 but no luck.

**Comment by bartwitkowski** [comment]
Hey @ms-zhenhua 
Will your fix solve my API version problem?

**Comment by ms-zhenhua** [comment]
Hi @bartwitkowski , yes, I am fixing it now.

**Comment by nvukas59** [comment]
Hello guys, 

I have the same issue when I'm trying to destroy deployment schedule on softwareUpdateConfigurations.
Do we have an update for the fix ? I saw it under review since 24 march...

**Comment by ms-zhenhua** [comment]
Hi @nvukas59 , the [PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/15758) is till under review.

**Comment by bartwitkowski** [comment]
[Fixed in version v3.15.0 (although I haven't checked it yet)](https://github.com/hashicorp/terraform-provider-azurerm/pull/15758#issuecomment-1193651131)

**Comment by bartwitkowski** [comment]
[Fixed in version v3.15.0 (although I haven't checked it yet)](https://github.com/hashicorp/terraform-provider-azurerm/pull/15758#issuecomment-1193651131)
